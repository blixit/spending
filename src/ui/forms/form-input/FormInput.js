import React from 'react';
import styled from 'styled-components';

import { Input } from 'ui/inputs/input/Input';
import Label from 'ui/forms/form-label/FormLabel';

const ErrorText = styled.div`
  color: red;
`;

const Container = styled.div`
  padding-top: 5px;
  padding-bottom: 5px;
`;

const FormInput = ({ label, error, type, onChange, arabic, ...rest }) => {
  const inputProps = { type, onChange, arabic, ...rest };
  return (
    <Container {...rest} >
      <Label arabic={arabic}>
        <span>{label}</span>
        <Input {...inputProps} />
      </Label>
      {error && <ErrorText>{error}</ErrorText>}
    </Container>
  )
};

FormInput.displayName = 'FormInput';

export default FormInput;
