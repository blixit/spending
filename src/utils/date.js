/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};