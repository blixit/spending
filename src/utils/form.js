/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
