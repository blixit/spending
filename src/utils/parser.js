export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);