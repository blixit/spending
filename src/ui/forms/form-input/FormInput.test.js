import render from 'testing/render';

import { withTheme } from 'ui/theming/test.utils';

import FormInput from './FormInput';

describe('[forms/form-input] FormInput', () => {
  it('renders properly', () => {
    const props = {
      label: 'Label',
      type: 'text',
      onChange: jest.fn()
    };
    const tree = render(withTheme(FormInput, props));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
