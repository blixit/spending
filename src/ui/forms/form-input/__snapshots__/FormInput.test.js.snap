// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[forms/form-input] FormInput renders properly 1`] = `
<div
  className="sc-bxivhb kKnJWp"
>
  <label
    className="sc-bwzfXH cJnpOh"
  >
    <span>
      Label
    </span>
    <input
      className="sc-bdVaJa dqOyiS"
      onChange={[Function]}
      type="text"
    />
  </label>
</div>
`;
